lib/devise_two_factorable/models/two_factorable.rb

```diff
@@ -95,7 +95,13 @@ def validate_otp_token(token, recovery = false)
 
     def validate_otp_time_token(token)
       return false if token.blank?
-      validate_otp_token_with_drift(token)
```

Comment:
The pattern is already here to support it, I would use the same guard pattern.
`return false unless validate_otp_token_with_drift(token)`
Also, since `self.save!` returns true on success and exception on failure, that turns this whole method into a lot less complexity:

```ruby
return false if token.blank?
return false unless validate_otp_token_with_drift(token)

self.last_successful_otp_at = Time.current

self.save!
```